### ISSUE TITLE
Label set in paper

### ISSUE THREAD

**Issue by bkj** [VSainteuf/pytorch-psetae#3]
In the paper, the results are reported for a 20-class classification task.  However, it looks like the number of classes in the dataset are either 17 or 35:
```
>>> z = json.load(open('labels.json'))
>>> len(set(z['label_19class'].values()))
17
>>> len(set(z['label_44class'].values()))
35
```

Are you able to give more details on the setup of the experiments in the paper?

Thanks!

__Edit:__ Also -- there's a `-1` class in `label_19class` -- what does that mean?

**Comment by VSainteuf** [comment]
- A bit of background: the annotated parcel database which we used to produce our labels provides a very fine grained nomenclature of ~400 classes.  Yet, *superclass* nomenclatures that group similar classes together in label sets with fewer elements are also provided. The two label files provided in our dataset correspond to two such superclass systems (the classes in the 19 class system being more high level than the 44 one). 

- Not all classes of the nomenclature are observed in the S2 tile (hence 17 and 35 instead of 19 and 44)

- In the paper we use the 44 class system, but we also select those classes for which more than 100 parcels are available in the dataset, reducing the number of classes to 20.

**Comment by VSainteuf** [comment]
And I am not sure about the -1 class, I'll have a closer look!

**Comment by bkj** [comment]
Oops -- the -1 thing was my mistake -- apologies!

**Comment by VSainteuf** [comment]
Good news ;)

**Comment by VSainteuf** [comment]
Last thing about the classes: we use the `sub_class` argument of `PixelSetData`  to only consider these 20 classes with enough samples.

**Comment by priti28192** [comment]
Hi,
Thank you very much for providing the implementation of your paper.
I am trying to understand your code. But, I am not able to find the names of the 20 or 44 classes in the order used in the implementation. Can you direct me towards the name of the classes.

Kind regards,
Priti

**Comment by VSainteuf** [comment]
0. Undefined
1. Meadow
2. Spring Durum Wheat
3. Winter Durum Wheat
4. Spring Cereal
5. Summer Cereal
6. Winter Cereal
7. Corn and Zea
8. Sorghum / Millet / Moha
9. Cereal
10. Buckwheat
11. Hemp
12. Leguminous Fodder
13. Other Fodder
14. Winter Rapeseed
15. Spring Rapeseed
16. Sunflower
17. Oleaginous
18. Soy
19. Protein Crop
20. Rice
21. Beetroot
22. Flax
23. Potato
24. Tobacco
25. Banana
26. Coffee / Cocoa
27. Dry Vegetable
28. Fruits, Vegetables, Flowers
29. Lavender
30. Fragrant Plants, Aromatic and Medicinal
31. Ligneous
32. Olive groves
33. Orchards
34. Grapevine
35. Short Rotation Coppice
36. Non-agricultural Surface
37. Built Area
38. Biomass
39. Wood
40. Sugar cane
41. Water, Marsh
42. Hop
43. Other

**Comment by VSainteuf** [comment]
Here's the list of class names for each label. Was that what you needed ?

**Comment by priti28192** [comment]
Thank you very much @VSainteuf  for your quick response. This is what I was looking for. Thank you for that. I had one more question. The pre-trained weights provided with the repo, is it trained on this label_44class nomenclature? Or you are using label_20class nomenclature for it. Also, is it possible to get access to the actual Sentinel-2 dataset with the timestamps? 

Kind regards,
Priti

**Comment by VSainteuf** [comment]
Yes, as said earlier in this issue:

> * Not all classes of the nomenclature are observed in the S2 tile (hence 17 and 35 instead of 19 and 44)
> * In the paper we use the 44 class system, but we also select those classes for which more than 100 parcels are available in the dataset, reducing the number of classes to 20.

So we use the 44 label set, reduced to the 20 most frequent classes.
And I don't have the original Sentinel-2 dataset anymore. You would need to re-download it using the timestamps of dates.json.

**Comment by priti28192** [comment]
Thanks @VSainteuf for the information. It was very helpful of you.

**Comment by nathanokole** [comment]
Hi,
One quick question: I can see you have a class named Beetroot, does this class also encompass sugar beet?
Also, the link to the pretrained weights does not work. Is it possible to have a working link?
Thanks,
Nathan
